src/reducers/composer-helpers.ts

```diff
@@ -22,49 +22,54 @@ import Console from 'global/console';
 
 const identity = state => state;
 
-/* eslint-disable no-unused-vars */
-// @ts-ignore
-export function log(text) {
+/** Returns a function that logs a value with a given message */
+export function log(text: string): (value: any) => void {
   return value => Console.log(text, value);
 }
-/* eslint-enable no-unused-vars */
-
-export function payload_(p) {
-  return {payload: p};
+/** Wraps a value in an object and stores it the `payload` field */
+export function payload_<P>(payload: P) {
+  return {payload};
 }
-
-export function apply_(updater, payload) {
+/** Wraps a value in an object and stores it the `payload` field */
+export function apply_<State, P>(
+  updater: (state: State, payload: P) => State,
+  payload: P
+): (state: State) => State {
   return state => updater(state, payload);
 }
 
-export function with_(fn) {
+export function with_<State>(
+  fn: (state: State) => (state: State) => State
+): (state: State) => State {
   return state => fn(state)(state);
 }
 
-export function if_(pred, fn) {
+export function if_<State>(pred: boolean, fn: (state: State) => State): (state: State) => State {
   return pred ? fn : identity;
 }
 
-export function compose_(fns) {
+export function compose_<State>(fns: Array<(s: State) => State>): (s: State) => State {
   return state => fns.reduce((state2, fn) => fn(state2), state);
 }
-
-export function merge_(obj) {
+/** Returns a reducer function that merges props with state */
+export function merge_<State, Props>(obj: Props): (state: State) => State & Props {
```

Comment:
You should instantiate the 2nd generic type in from of the function it is first used. so `merge_` is better defined this way
```js
export function merge_<Props>(obj: Props): <State>(state: State) => State {
  return state => ({...state, ...obj});
}
```